src/main/java/com/develhope/spring/users/controllers/UserController.java

```diff
@@ -14,4 +17,12 @@ public class UserController {
     @Autowired
     private UserService userService;
 
+
+    @PatchMapping ("/{userID}")
+    @ResponseStatus(HttpStatus.OK)
+    @Secured("ADMIN")
+    public UserReworkedDTO updateUser(@PathVariable long userID,
```

Comment:
Essendo che Il venditore può modificare tutti, pure i venditori e può creare un venditore. Potremmo fare che questo endpoint può essere usato per prendere 3 piccioni con una fava, cioè potresti modificare questo updaterDTO in modo che modifichi anche il ruolo, in modo che questo end point possa essere usato per modificare un qualsiasi utente, modificare un venditore, e creare un venditore a partire un utente account semplicemente impsotando il ruolo su venditore. Poi rimarrebbe fuori l end point per l'user che possa modificare se stesso. Se trovi il modo di inserirlo nello stesso endopoint richiamando due service diversi ben venga, se no probabilmente vanno separati. Intanto però io farei questa modifica, così ti fai tre task in uno

Reply:
Oddio, non so se convenga spostare un buyer a venditore, a livello aziendale conviene tenere gli account ben separati

Reply:
Cioè, si potrebbe fare, tanto non è roba che verrà lanciata sul mercato